### PR TITLE
Adding exteralize string support for i18n

### PR DESCRIPTION
**Pull request by upkarlidder**
**This is still work in progress. Please do not merge.** @davidnixon, @GeraldMit please review when you get a chance.

- [x] add a way to externalize strings in the vue app
- [x] add language support with the [vue-i18n](https://kazupon.github.io/vue-i18n/) package
- [x] add language picker with `vue-flag-icon`

**The languages files were generated with ~~Google Translate~~ [Watson Translation service](https://www.ibm.com/demos/live/watson-language-translator/self-service/home) and should not be added to production in their current state.**

Externalize all pages and components
- [x] ConnectPage
- [x] JourneyPage
- [x] LandingPage
- [x] RatePage
- [x] SupportPage
- [x] WhyVotePage

The labels in the countdown widget are currently fixed, but there is a [PR](https://github.com/getanwar/vuejs-countdown/pull/14) to make them dynamic. We can externalize the labels once it is merged in the original widget.

Landing page in Hindi. The language picker on top is just a place holder to test. 

image

### CODE REVIEW

File: ui/src/plugins/es.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  "appHeaderWhyVote": "Por qué votar",
+  "appHeaderVoterJourney": "Viaje de votantes",
+  "appHeaderGetConnected": "Cómo conectarse",
+  "appHeaderVoterSupport": "Apoyo electoral",
+  "countdownHeader": "Elecciones legislativas intermedias",
+  "landingPageMain": "Nuestra Misión",
+  "landingPageDesc": "Con cinco quintos de votante, ayudamos a poner el poder en sus manos. Nuestra plataforma proporciona una experiencia personalizada con herramientas útiles para ayudar a los votantes marginados a superar los bloqueos de proceso de votación. Cinco quintos de votante tiene en cuenta varios viajes de votación, como los votantes por primera vez, los veteranos, las minorías, los discapacitados legales, los ancianos, o aquellos con condenas graves. Juntos, podemos cambiar la narrativa de las personas históricamente marginadas en nuestro proceso democrático.",
+  "landingPageEdu": "Educar.",
+  "landingPageQuote": "\"El conocimiento es poder.\"- Francis Bacon",
+  "landingPageEduDesc": "Cinco quintos de los votantes educarán a los votantes en derechos de voto con la guía de votación y la conexión a las herramientas actualizadas necesarias para completar el proceso de votación.",
```
Comment (GeraldMit) [inline ui/src/plugins/es.json:14]: Product name should be extracted? possibly reword or have substitution character sequence in strings? 
cinco quintos de votante translation also lost Five Fifths Voter from brand/product into the statement.

File: ui/src/plugins/es.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  "landingPageEmpower": "Potenciador.",
+  "landingPageEmpowerQuote": "\"No soy pájaro, y ninguna red me ensanza: soy un ser humano libre con una voluntad independiente. \" - Charlotte Brontë",
```
[the remark this answers is not in the document]
Comment (GeraldMit) [reply ui/src/plugins/es.json:16]: for hi.json, the quotes are not translated.

File: ui/src/plugins/es.json (hunk 1 above)
Comment (GeraldMit) [reply ui/src/plugins/es.json:14]: I am guessing that this rephtasing needs to be considered with the next wave of development effort.

**Review by GeraldMit — APPROVED** [review]
Addrd comments for consideration but approve generally.

File: ui/src/plugins/es.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (upkarlidder) [reply ui/src/plugins/es.json:16]: For the first part, can you please help to translate all the quotes with Watson NLU?

[the remark this answers is not in the document]
Comment (upkarlidder) [reply ui/src/plugins/es.json:16]: The current translations are just samples. I was not going to translate all the strings.

File: ui/src/plugins/es.json (hunk 1 above)
Comment (upkarlidder) [reply ui/src/plugins/es.json:14]: Yes, thank you @GeraldMit. We can use this - https://vue-i18n.intlify.dev/guide/essentials/syntax.html#named-interpolation

Is the product name translated at all? Taking hindi as example:

1. फाइव फिफ्थ वोटर. This reads as Five Fifth Voters, but the words do not mean anything in Hindi.
2. पांचवां मतदाता does not read Five Fifth Voters, but means five fifth voters when translated to English.
3. Leave as Five Fifth Voters. As in `Five Fifth Voter के साथ, हम शक्ति को वापस अपने हाथों में रखने में ..."

Comment (upkarlidder) [reply ui/src/plugins/es.json:14]: I went with `#3` for now.
image

**Comment by upkarlidder** [comment]
Externalized strings for a few more pages. Still have a  few pages to go.

**Comment by aleggett23** [comment]
Hi team! This looks great! This is so touching because there is now an opportunity to reach and help even more individuals by tackling those language barriers🦾💪🏾🙌🏾

I agree that in terms of deciding whether to translate the Five Fifths Voter name, it may be easier from a dev standpoint, more consistent, and more fluid when connecting different cultures together from a brand building perspective to keep it Five Fifths Voter throughout as is for now like in #3. 

I was thinking about brands like Facebook, Twitter, and other platforms for example that have language options/international versions but are still known by the original name for the most part. The name itself also has several underlying meanings, so direct translations of it through multiple languages may prove to make some confusion down the road. I think too when people refer the product in whatever culture or demographic, that keeping the same core name may promote a common point of reference, even when the site is translated. 

Is this what you were expressing in #3 @upkarlidder @GeraldMit ? Also with the quotes, I added those, and can also be of help with determining how it may be most user-friendly if need be.

**Comment by sebastianhuynh-nyc** [comment]
Hi, I do agree that the name should be kept as it is, with that said, we need to explore a formal branding for the solution.
Upkar, lets find time to chat and see what we can do now and what we want to do in the mid term.

 
 
 
 
 
Sebastian Huynh
Design Lead
Industry Marketing
 
IBM ORIGINALS
 

 
 
Sebastian Huynh
E-mail: ***@***.***
 
 
----- Original message -----From: "aleggett23" ***@***.***>To: "Call-for-Code-for-Racial-Justice/Five-Fifths-Voter" ***@***.***>Cc: "sebastianhuynh-nyc" ***@***.***>, "Assign" ***@***.***>Subject: [EXTERNAL] Re: [Call-for-Code-for-Racial-Justice/Five-Fifths-Voter] WIP - adding exteralize string support for i18n (#192)Date: Wed, Aug 25, 2021 10:12 AM      Hi team! This looks great! This is so touching because there is now an opportunity to reach and help even more individuals by tackling those language barriers 🦾 💪🏾🙌🏾 I agree that in terms of deciding whether to translate the Five Fifths Voter    ZjQcmQRYFpfptBannerStart  

This Message Is From an External Sender
This message came from outside your organization. ZjQcmQRYFpfptBannerEnd 
 
Hi team! This looks great! This is so touching because there is now an opportunity to reach and help even more individuals by tackling those language barriers🦾💪🏾🙌🏾
I agree that in terms of deciding whether to translate the Five Fifths Voter name, it may be easier from a dev standpoint, more consistent, and more fluid when connecting different cultures together from a brand building perspective to keep it Five Fifths Voter throughout as is for now like in #3.
I was thinking about brands like Facebook, Twitter, and other platforms for example that have language options/international versions but are still known by the original name for the most part. The name itself also has several underlying meanings, so direct translations of it through multiple languages may prove to make some confusion down the road. I think too when people refer the product in whatever culture or demographic, that keeping the same core name may promote a common point of reference, even when the site is translated.
Is this what you were expressing in #3 @upkarlidder @GeraldMit ? Also with the quotes, I added those, and can also be of help with determining how it may be most user-friendly if need be.

**Comment by krook** [comment]
Also a +1 on keeping "Five Fifths Voter" as the universal name. 

Perhaps a stylized icon or logo without the text (maybe incorporating the numbers 5/5) can achieve some degree of extra internationalization?

**Comment by aleggett23** [comment]
Sounds good @sebastianhuynh-nyc & Great idea @krook! The team has been mentioning a logo idea for a while! It was a GitHub issue at one point but not sure if it still is.

Maybe deciding upon a logo could be a portion of the formal branding process that @sebastianhuynh-nyc is referring to.

**Comment by aleggett23** [comment]
@upkarlidder Thank you so much for your hard work, and absolutely!  I will message you about potential options.

**Comment by upkarlidder** [comment]
@GeraldMit @davidnixon can you please merge when you get a chance. I am done with this PR.

@davidnixon I assume this will kick off a build and reflect on the production site? I took out the language selection, but left the translated files in there.

**Comment by upkarlidder** [comment]
@GeraldMit @davidnixon can you please merge when you get a chance. I am done with this PR.

File: ui/src/components/BallotReturn/BallotReturn.vue [hunk 3]
```diff
@@ -47,7 +45,10 @@
                 </cv-list-item>
               </cv-list>
 
-              <span><br />Powered by the Civic Information API</span>
+              <span
```
Comment (davidnixon) [inline ui/src/components/BallotReturn/BallotReturn.vue:48]: I don't think this one should be internationalized. It's a required string that has to be displayed when we use these results.

File: ui/src/App.vue [hunk 4]
```diff
@@ -1,5 +1,10 @@
 <template>
   <div id="app">
+    <!-- <div>
+      <button v-for="entry in languages" :key="entry.title" @click="changeLocale(entry.language)">
+        <flag :iso="entry.flag" v-bind:squared="false" /> {{ entry.title }}
+      </button>
+    </div> -->
```
Comment (davidnixon) [inline ui/src/App.vue:7]: I don't think you want any of this here. I think you want to use it in `ui/src/components/AppHeader/AppHeader.vue`
You can see here there is place to add things to the header bar. The user avatar is already there.

https://github.com/Call-for-Code-for-Racial-Justice/Five-Fifths-Voter/blob/e196f0c4fee5c50ef55f358330c777106ab2c675/ui/src/components/AppHeader/AppHeader.vue#L24-L28

https://vue.carbondesignsystem.com/?path=/story/components-cvuishell--header-base-with-navigation-and-actions

You would also need to set a value for each flag in the data component. 
```javascript
{ flag: 'us', language: 'en', title: 'English' },
```

File: ui/src/plugins/i18n.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  kr: kr,
+  ru: ru,
+  sc: sc,
+  zh: zh,
+  ar: ar,
+  vn: vn
+};
+
+export default new VueI18n({
+  locale: 'en',
```
Comment (davidnixon) [inline ui/src/plugins/i18n.js:28]: I think you should set the default locale based on the browser language. 

https://github.com/kazupon/vue-i18n/issues/220

I have my firefox browser (not my ina browser) enabled for Spanish, English, & German to test internationalization. You should try that.

**Review by davidnixon — CHANGES_REQUESTED** [review]
This is really nice work! Thank you so much.

Also, I'm not sure if this PR is meant to surface the language switcher or not. Maybe that's a future work?

File: ui/src/App.vue (hunk 4 above)
Comment (upkarlidder) [reply ui/src/App.vue:7]: Thank you @davidnixon. It's commented out right now. I will move it to the user avatar section in a different PR.

File: ui/src/components/BallotReturn/BallotReturn.vue (hunk 3 above)
Comment (upkarlidder) [reply ui/src/components/BallotReturn/BallotReturn.vue:48]: Reverted back. Thank you @davidnixon

**Review by davidnixon — APPROVED** [review]

**Pull request merged**